src/main/java/com/google/api/codegen/metacode/FieldStructureParser.java

```diff
@@ -190,12 +181,28 @@ public static InitCodeNode parsePath(InitCodeNode root, Scanner scanner) {
     }
   }
 
-  private static String parseValue(Scanner scanner) {
+  private static String parseKey(Scanner scanner) {
     int token = scanner.scan();
     Preconditions.checkArgument(
         token == Scanner.INT || token == Scanner.IDENT || token == Scanner.STRING,
         "invalid value: %s",
         scanner.input());
     return scanner.tokenStr();
   }
+
+  /**
+   * Parses the RHS of configs. If the RHS is a string literal we return the unquoted string.
+   * Otherwise we strip leading spaces and return the rest of RHS as a string for backward
+   * compatibility.
+   */
+  private static String parseValue(Scanner scanner) {
+    int token = scanner.scan();
+    if (token == Scanner.STRING) {
```

Comment:
I would add a comment here reminding readers that `Scanner.STRING` means a double-quoted string literal.